lib/grandpa/network.go

```diff
@@ -64,8 +64,8 @@ func (*GrandpaHandshake) Type() byte {
 }
 
 // Hash ...
-func (*GrandpaHandshake) Hash() common.Hash {
-	return common.Hash{}
+func (*GrandpaHandshake) Hash() (common.Hash, error) {
+	return common.Hash{}, nil
```

Comment:
for all these message types not hashing anything, should we `panic("unimplemented")` to make sure these Hash are never called?

Reply:
linked this comment in the issue for refactoring opportunities https://github.com/ChainSafe/gossamer/issues/2437